src/gen/coin_id.rs

```diff
@@ -6,12 +7,12 @@ pub fn compute_coin_id(
     parent_id: NodePtr,
     puzzle_hash: NodePtr,
     amount: &[u8],
-) -> [u8; 32] {
+) -> Bytes32 {
```

Comment:
I think it's better to avoid non-native types where possible. Let the caller of this function use `.into`.

Reply:
(This should be weighed against lots of `.into` calls that might otherwise not be needed.)

Reply:
what's your rationale for that? I would think having even more specific types (like, `CoinId`, `PuzzleHash` etc.) would improve type safety.

I suppose the main purpose of `BytesN` types is to provide python conversions, and this function isn't exposed to python, so technically doesn't need it.

It's called exactly once in production code (to construct an `Arc<Bytes32>`) + a few times in the unit test. Does this preference also extend to the `coin_id` member of the `Spend` class?

Reply:
I'm pretty sure `pyo3` knows how to convert `&[u8]` to `PyBytes` already, so re-implementing for `Bytes32` probably isn't even necessary. My concern is a very mild one about polluting namespace and dependency-space here with a custom `Bytes32` type in this file when it probably isn't strictly necessary.

Reply:
It's necessary for supporting serialization, to have a type to implement the `Serialize` and `Deserialize` traits. This specific return value isn't covered by that either, but I believe the members it initialized are.